packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -835,7 +872,33 @@ class _RenderSegmentedButton<T> extends RenderBox with
     return Size(childWidth, maxHeight);
   }
 
+  Size _calculateVerticalChildSize(BoxConstraints constraints) {
+    double maxWidth = 0;
+    RenderBox? child = firstChild;
+    double childHeight;
+    if (_isExpanded) {
+      childHeight = constraints.maxHeight / childCount;
+    } else {
+      childHeight = constraints.minHeight / childCount;
+      while (child != null) {
+        childHeight = math.max(childHeight, child.getMaxIntrinsicHeight(double.infinity));
+        child = childAfter(child);
+      }
+      childHeight = math.min(childHeight, constraints.maxWidth / childCount);
```

Comment:
I think this should also be `constraints.maxHeight / childCount`? Because if direction is vertical, we should consider the height of the constraints instead of width, right?

Reply:
I can see my code can lead to this bad behavior :) when the parent height is less than the `segmentedButton` height , and can lead to hide some items in the list , now i can confirm this, thanks Qun.